utils/src/main/java/com/cloud/utils/SwiftUtil.java

```diff
@@ -236,4 +247,60 @@ public static boolean deleteObject(SwiftClientCfg cfg, String path) {
         command.execute(parser);
         return true;
     }
+
+    public static boolean setTempKey(SwiftClientCfg cfg, String tempKey){
+
+        Map<String, String> tempKeyMap = new HashMap<>();
+        tempKeyMap.put("Temp-URL-Key", tempKey);
+        return postMeta(cfg, "", "", tempKeyMap);
+
+    }
+
+    public static URL generateTempUrl(SwiftClientCfg cfg, String container, String object, String tempKey, int urlExpirationInterval) {
+
+        int currentTime = (int) (System.currentTimeMillis() / 1000L);
+        int expirationSeconds = currentTime + urlExpirationInterval;
+
+        try {
+
+            URL endpoint = new URL(cfg.getEndPoint());
+            String method = "GET";
+            String path = String.format("/v1/AUTH_%s/%s/%s", cfg.getAccount(), container, object);
+
+            //sign the request
+            String hmacBody = String.format("%s\n%d\n%s", method, expirationSeconds, path);
+            String signature = calculateRFC2104HMAC(hmacBody, tempKey);
+            path += String.format("?temp_url_sig=%s&temp_url_expires=%d", signature, expirationSeconds);
+
+            //generate the temp url
+            URL tempUrl = new URL(endpoint.getProtocol(), endpoint.getHost(), endpoint.getPort(), path);
+
+            return tempUrl;
+
+        } catch (Exception e) {
+            logger.error("Unable to generate temperoary URL " + e.getMessage());
```

Comment:
Should log the exception, not just the message. And why catch all exceptions? What exceptions are you expecting?